### DISCUSSION TITLE
TCStat valid time output

### DISCUSSION THREAD

**Discussion by mollybsmith-noaa** [dtcenter/METplus#1648] [Configuration] [answered]
Hi all,

Last year, George McCabe added an option to TCPairs (TC_PAIRS_WRITE_VALID) to only output stats for a specific valid time. For example, if you were running TCPairs on model cycles from 6/28 to 7/08, you could configure TCPairs to only output stats that were valid at 7/09. So you'd get the results for the 24 hr forecast on 7/08, the 48 hr forecast on 7/07, the 72 hr forecast on 7/06, etc, even though TCPairs was actually verifying all valid times and forecasts leads behind the scenes. Would it be possible to get the same functionality added to TCStat as well? This option has been really valuable for verifying in real time and now getting duplicate data in the database. 

Thanks!

**Comment by CPKalb** [comment, marked as answer]
Hi Molly:

Tc Stat should already able to filter by valid time.  This can be done using the valid_inc option either on a command line job or in the TCStat config file.  Here is an example:

tc_stat -lookin MET_test_output/tc_pairs/alal2010_all.tcst -job filter -dump_row out.tcst -valid_inc 20100813_000000

This job would read data from the input tcst file and retain lines whose valid time is 20100813_000000.  It would then write them to an output tcst file as set using the [TC_STAT_VALID_INCLUDE](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html?highlight=WRITE_VALID#term-TC_STAT_VALID_INCLUDE) option.

Will this work for your needs?

Christina

**Comment by mollybsmith-noaa** [comment]
I think that is what I'm looking for. With TCPairs, George said that I needed both TC_PAIRS_VALID_INCLUDE and TC_PAIRS_WRITE_VALID. At the time I don't think the functionality did exist in TCStat yet. Glad to know that it does now, but do I need only TC_STAT_VALID_INCLUDE or is there also a TC_STAT_WRITE_VALID? Thanks!

**Reply by CPKalb** [reply]
There is not a TC_STAT_WRITE_VALID.  You should just need the TC_STAT_VALID_INCLUDE.

Thanks

**Reply by mollybsmith-noaa** [reply]
Okay, I'll give it a try. Thanks!
